src/Jellyfin.Networking/AutoDiscoveryHost.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.Net;
+using System.Net.Sockets;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using MediaBrowser.Common.Configuration;
+using MediaBrowser.Common.Net;
+using MediaBrowser.Controller;
+using MediaBrowser.Model.ApiClient;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace Jellyfin.Networking;
+
+/// <summary>
+/// <see cref="BackgroundService"/> responsible for responding to auto-discovery messages.
+/// </summary>
+public sealed class AutoDiscoveryHost : BackgroundService
+{
+    /// <summary>
+    /// The port to listen on for auto-discovery messages.
+    /// </summary>
+    private const int PortNumber = 7359;
+
+    private readonly ILogger<AutoDiscoveryHost> _logger;
+    private readonly IServerApplicationHost _appHost;
+    private readonly IConfigurationManager _configurationManager;
+    private readonly INetworkManager _networkManager;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="AutoDiscoveryHost" /> class.
+    /// </summary>
+    /// <param name="logger">The <see cref="ILogger{AutoDiscoveryHost}"/>.</param>
+    /// <param name="appHost">The <see cref="IServerApplicationHost"/>.</param>
+    /// <param name="configurationManager">The <see cref="IConfigurationManager"/>.</param>
+    /// <param name="networkManager">The <see cref="INetworkManager"/>.</param>
+    public AutoDiscoveryHost(
+        ILogger<AutoDiscoveryHost> logger,
+        IServerApplicationHost appHost,
+        IConfigurationManager configurationManager,
+        INetworkManager networkManager)
+    {
+        _logger = logger;
+        _appHost = appHost;
+        _configurationManager = configurationManager;
+        _networkManager = networkManager;
+    }
+
+    /// <inheritdoc />
+    protected override Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        if (!_configurationManager.GetNetworkConfiguration().AutoDiscovery)
+        {
+            return Task.CompletedTask;
+        }
+
+        // Linux needs to bind to the broadcast addresses to receive broadcast traffic
+        if (OperatingSystem.IsLinux())
+        {
+            _ = Task.Run(() => ListenForAutoDiscoveryMessage(IPAddress.Broadcast, stoppingToken), stoppingToken);
+        }
+
+        foreach (var intf in _networkManager.GetInternalBindAddresses())
+        {
+            var address = OperatingSystem.IsLinux()
+                ? NetworkUtils.GetBroadcastAddress(intf.Subnet)
+                : intf.Address;
+
+            _ = Task.Run(() => ListenForAutoDiscoveryMessage(address, stoppingToken), stoppingToken);
+        }
+
+        return Task.CompletedTask;
```

Comment:
You should return all created tasks so that "StopAsync" will await them when stop is requested to ensure they are properly stopped

https://learn.microsoft.com/en-us/aspnet/core/fundamentals/host/hosted-services?view=aspnetcore-8.0&tabs=visual-studio#backgroundservice-base-class